### PR TITLE
[KOGITO-66] - Add ServiceAccount to the KogitoApp

### PR DESCRIPTION
**Pull request by ricardozanini**
This PR includes:

1. Operator SDK upgrade to 0.9.0
2. Openapi auto generation of CRDs
3. KogitoApp pods now run with a specified ServiceAccount that can fetch for Kubernetes API Resources within the namespace (like services)

### CODE REVIEW

**Comment by sutaakar** [comment]
@ricardozanini I have noticed that some descriptions were deleted from CRD, was it intentional?

**Comment by ricardozanini** [comment]
> @ricardozanini I have noticed that some descriptions were deleted from CRD, was it intentional?

No, I didn't realized  that. The CRD is now being generated automatically, so those descriptions are taken from the comments on `*_types.go` files. Thanks for pointing that out, I'm going to add them.

**Comment by ricardozanini** [comment]
@tchughesiv could you take a look when you have the time, please? :nerd_face:

File: pkg/apis/app/v1alpha1/kogitoapp_types.go [hunk 1]
```diff
@@ -138,7 +182,7 @@ type KogitoApp struct {
 	metav1.TypeMeta   `json:",inline"`
 	metav1.ObjectMeta `json:"metadata,omitempty"`
 
-	Spec   KogitoAppSpec   `json:"spec,omitempty"`
+	Spec   *KogitoAppSpec  `json:"spec"`
```
Comment (tchughesiv) [inline pkg/apis/app/v1alpha1/kogitoapp_types.go]: could you explain why this is being made a pointer? i don't see any corresponding code to indicate a reason for the change changes which is why i ask

File: pkg/controller/kogitoapp/kogitoapp_controller.go [hunk 2]
```diff
@@ -431,6 +455,40 @@ func (r *ReconcileKogitoApp) newDCForCR(cr *v1alpha1.KogitoApp, serviceBC obuild
...
+
+func newSARoleBinding(cr *v1alpha1.KogitoApp, sa corev1.ServiceAccount) (roles rbacv1.RoleBindingList) {
+	roles = rbacv1.RoleBindingList{}
+	roles.Items = append(roles.Items, rbacv1.RoleBinding{
+		ObjectMeta: metav1.ObjectMeta{
+			Namespace: sa.Namespace,
+			Name:      fmt.Sprintf("%s-%s", sa.Name, constants.ServiceAccountRole),
+		},
+		RoleRef: rbacv1.RoleRef{
+			Kind: "ClusterRole",
```
Comment (tchughesiv) [inline pkg/controller/kogitoapp/kogitoapp_controller.go]: is the intention here to give all kogito deployments `view` rights to the cluster? if so, this seems excessive. what is the purpose of this change?

File: pkg/apis/app/v1alpha1/kogitoapp_types.go (hunk 1 above)
Comment (ricardozanini) [reply pkg/apis/app/v1alpha1/kogitoapp_types.go]: the pointer makes the spec required in the crd. But thinking again, is a little too excessive. I believe that there's no need to do that.

File: pkg/controller/kogitoapp/kogitoapp_controller.go (hunk 2 above)
Comment (ricardozanini) [reply pkg/controller/kogitoapp/kogitoapp_controller.go]: this is the role that we've been testing with the service discovery mechanism to access the Core Services API. Sometimes, this services are in other namespaces, that's why we need a `ClusterRole`. Makes sense?

**Comment by tchughesiv** [comment]
@ricardozanini do the operator-sdk and go version requirements need to change in the readme? circleci updated?

**Comment by ricardozanini** [comment]
> @ricardozanini do the operator-sdk and go version requirements need to change in the readme? circleci updated?

I changed the circleci when we upgraded the GO version in other PR. The operator-sdk doesn't matter for it.

**Comment by tchughesiv** [comment]
LGTM
